packages/contracts-periphery/contracts/universal/op-nft/OptimistInviter.sol

```diff
@@ -0,0 +1,274 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Semver } from "@eth-optimism/contracts-bedrock/contracts/universal/Semver.sol";
+import { AttestationStation } from "./AttestationStation.sol";
+import { SignatureChecker } from "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import {
+    EIP712Upgradeable
+} from "@openzeppelin/contracts-upgradeable/utils/cryptography/draft-EIP712Upgradeable.sol";
+
+/**
+ * @custom:upgradeable
+ * @title  OptimistInviter
+ * @notice OptimistInviter issues "optimist.can-invite" and "optimist.can-mint-from-invite"
+ *         attestations. Accounts that have a "optimist.can-invite" attestation can issue
+ *         signatures that allow other accounts to claim an invite. The invitee uses a claim
+ *         and reveal flow to claim the invite to an address of their choosing.
+ *
+ *         Parties involved:
+ *           1) INVITE_GRANTER: trusted account that can allow accounts to issue invites
+ *           2) issuer: account that is allowed to issue invites
+ *           3) claimer: account that receives the invites
+ *
+ *         Flow:
+ *           1) INVITE_GRANTER calls _setInviteCount to allow an issuer to issue a certain number
+ *              of invites, creating "optimist.can-invite" attestations for the issuer
+ *           2) Off-chain, the issuer signs (EIP-712) a ClaimableInvite to produce a signature
+ *           3) Off-chain, invite issuer sends the plaintext ClaimableInvite and the signature
+ *              to the recipient
+ *           4) claimer chooses an address they want to receive the invite on
+ *           5) claimer commits the hash of the address they want to receive the invite on and the
+ *              received signature keccak256(abi.encode(addressToReceiveTo, receivedSignature))
+ *              using the commitInvite function
+ *           6) claimer reveals the plaintext ClaimableInvite and the signature using the
+ *              claimInvite function, receiving the "optimist.can-mint-from-invite" attestation
+ */
+contract OptimistInviter is Semver, EIP712Upgradeable {
+    /**
+     * @notice Emitted when an invite is claimed.
+     *
+     * @param issuer Address that issued the signature.
+     * @param claimer Address that claimed the invite.
+     */
+    event InviteClaimed(address indexed issuer, address indexed claimer);
+
+    /**
+     * @notice EIP712 typehash for the ClaimableInvite type.
+     */
+    bytes32 public constant CLAIMABLE_INVITE_TYPEHASH =
+        keccak256("ClaimableInvite(address issuer,bytes32 nonce)");
+
+    /**
+     * @notice Attestation key for granting invites.
+     */
+    bytes32 public constant CAN_INVITE_ATTESTATION_KEY = bytes32("optimist.can-invite");
```

Comment:
good point- For this contract it's a quick change.

But we already issued some attestations using non-hashed keys for the mint attestation for the [OptimistNFT contract](https://github.com/ethereum-optimism/optimism/blob/91f81d9976a1c0af27fb40a9d4f56c3315791808/packages/contracts-periphery/contracts/universal/op-nft/Optimist.sol#L114-L119). Prob too late to change, because we've already issued them, but going forward the policy should be to use hashed keys whenever possible.

Reply:
The atst sdk and our documentation both hash if over 32 and don't hash if under 32.   If you make it a best practice to always hash please update both sdk and docs to represent that cc @qbzzt

Reply:
i think we should only hash if len > 32, this is how many things in ethereum consensus work, like for example leaves in merkle trees that are in the block header

Reply:
solidity also hashes things that are abitrary len when their size is greater than 31 to determine storage slots

Reply:
Got it - since this is a common accepted pattern, will stick to the policy of only hashing if a value is len > 32.